specification/confidentialledger/Microsoft.mCCF/service_state.tsp

```diff
@@ -0,0 +1,383 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+
+import "./common.tsp";
+
+@TypeSpec.Versioning.useDependency(Azure.Core.Versions.v1_0_Preview_2)
+namespace ServiceState;
+
+/// Common types
+@doc("PEM encoding of a cryptographic identifier. Contains a base64-encoded payload wrapped in content type identifiers.")
+scalar pem extends string;
+
+/// Member types
+@doc("Possible states for a CCF governing member.")
+enum MemberStatus {
+  @doc("Member has been approved to join the service, but have not yet submitted an ACK to indicate their participation. These members should not be considered by constitutions in quorum calculations, and their signed requests will be rejected until they become Active.")
+  Accepted,
+
+  @doc("Member has submitted a valid ACK and is a full voting participant.")
+  Active,
+}
+
+@doc("Information on individual members within a consortium.")
+@TypeSpec.Rest.resource("service/members")
+model Member {
+  @doc("ID of CCF member who this object refers to.")
+  @visibility("read")
+  @key
+  memberId: memberId;
+
+  @doc("Current status of this member.")
+  status: MemberStatus;
+
+  @doc("Arbitrary service-defined metadata about this member. May be used by constitution or application code, but will not affect any core framework decisions.")
+  memberData: {};
+
+  @doc("x509 certificate used as this member's identity.")
+  certificate: pem;
+}
+
+/// Node types
+@doc("Lifecycle state of a CCF node. Nodes will generally start as Pending, then transition to Trusted, then to Retired. They are only full participants in the service while they are Trusted.")
+enum NodeStatus {
+  @doc("Indicates a node which has successfully completed a join request to the service (including attestation checks), but has not yet been approved by governance.")
+  Pending,
+
+  @doc("Indicates a node which has been approved by governance, and is a full participant in the service.")
+  Trusted,
+
+  @doc("Indicates a node which has been removed from a service by a reconfiguration.")
+  Retired,
+}
+
+@doc("Common type for attestation information, describing the cryptographically-endorsed claim of what code is executing, and what platform it is executing on. Derived types contain platform-specific details.")
+@discriminator("format")
+model QuoteInfo {
+  @doc("Names the platform and version of the contained info.")
+  format: string;
+}
+
+@doc("Attestation information for Intel SGX enclaves.")
+model SgxQuoteInfo {
+  @doc("Fixed identifier for SGX information.")
+  format: "OE_SGX_v1";
+
+  @doc("Base-64 encoded SGX quote.")
+  quote: string;
+
+  @doc("Base-64 encoded SGX endorsements.")
+  endorsements: string;
```

Comment:
Should be of type `bytes`, though I don't recall in TypeSpec if you need a `@format(base64)`. The code generators will do the right thing instead of requiring the caller to encode/decode themselves.